include/dynd/types/tuple_type.hpp

```diff
@@ -22,15 +22,14 @@ namespace ndt {
     intptr_t m_field_count;
 
     /**
-     * Immutable contiguous array of field types. Always has type "N * type".
+     * Immutable vector of field types. Always has type "N * type".
      */
     const std::vector<type> m_field_types;
 
     /**
-     * Immutable contiguous array of arrmeta offsets. Always has type "N *
-     * intptr".
+     * Immutable vector of arrmeta offsets. Always has type "N * uintptr".
      */
-    nd::array m_arrmeta_offsets;
+    std::vector<uintptr_t> m_arrmeta_offsets;
 
```

Comment:
Yes, definitely.  The showstopper for now is https://github.com/libdynd/libdynd/pull/918/files#diff-d1a413db4b4cdcb4584670f4f43742caL417 (and also the initialization in the constructor should move to a function).